cmd/osmosisd/cmd/root.go

```diff
@@ -354,7 +355,8 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 		WithHomeDir(homeDir).
 		WithViper("OSMOSIS")
 
-	tempApp := osmosis.NewOsmosisApp(log.NewNopLogger(), cosmosdb.NewMemDB(), nil, true, map[int64]bool{}, osmosis.DefaultNodeHome, 5, sims.EmptyAppOptions{}, osmosis.EmptyWasmOpts, baseapp.SetChainID("osmosis-1"))
+	tempDir := fmt.Sprintf("%s%d", ".temp-osmosis", rand.Int())
```

Comment:
Needed this as we call this twice during basically every run of `osmosisd` it's necessary because of this new feature in wasmvm => https://github.com/CosmWasm/wasmvm/blob/main/internal/api/lib.go#L44-L63

Reply:
😢